[docs] Migrate Output API and Output functions

kabalin opened the pull request:
Migrated pages:
- https://docs.moodle.org/dev/Output_API
- https://docs.moodle.org/dev/Output_functions

Closes #179

<a href="https://gitpod.io/#https://github.com/moodle/devdocs/pull/216"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

andrewnicols commented on docs/apis/subsystems/output.md:
```diff
@@ -0,0 +1,281 @@
+
+<!-- cspell:ignore renderables, sometext, courseid, enabletrusttext -->
+
+## Overview
```
I'd recommend taking this out. We've standardised to remove the first title if it's something like "Introduction", or "Overview". There is no benefit.

andrewnicols commented on docs/apis/subsystems/output.md:
```diff
@@ -0,0 +1,281 @@
+echo $output->render($renderable);
+
+echo $output->footer();
+
```
```suggestion
```

andrewnicols commented on docs/apis/subsystems/output.md:
````diff
@@ -0,0 +1,281 @@
+
+```
+
+Explaining this demo step by step we have firstly the standard config and includes, plus the `admin_externalpage_setup('tooldemo')` which calls `require_login` and performs permissions checks for admin pages.
+
+```php
+require_once(__DIR__ . '/../../../config.php');
+require_once($CFG->libdir.'/adminlib.php');
+
+admin_externalpage_setup('tooldemo');
+```
+
+Then we load the title of the page from a lang string (see [String API](https://docs.moodle.org/dev/String_API) ). We use this string to set some `$PAGE` properties (URL title and heading).
+
+```php
+// Set up the page.
+$title = get_string('pluginname', 'tool_demo');
+$pagetitle = $title;
+$url = new moodle_url("/admin/tool/demo/index.php");
+$PAGE->set_url($url);
+$PAGE->set_title($title);
+$PAGE->set_heading($title);
+```
+
+:::note What is $PAGE and where did it come from ?
+
+`$PAGE` is a global variable used to track the state of the page that is being returned. It is an instance of the `moodle_page` class defined in `lib/pagelib.php`. See [Page API](https://docs.moodle.org/dev/Page_API) for more information on the `$PAGE` variable.
+
+:::
````
This is largely irrelevant to the Output API docs.

I'd suggest putting it into a collapsed `<details>` section or an `:::info` section, for example:

```suggestion
:::info Setup of an admin page

On admin pages the `admin_externalpage_setup($sectionname);` function should be used to setup and perform permission checks, for example:

```php title="admin/tool/demo/mypage.php"
require_once(__DIR__ . '/../../config.php');
require_once("{$CFG->libdir}/adminlib.php");

admin_externalpage_setup('example');
```

:::

```

I'd suggest detailing the PAGE API still, but with more info.

andrewnicols reviewed (changes requested):
I don't have a chance to finish a review now as I'm done for the day, but I have a couple of notes so far.

andrewnicols commented:
Oh, I meant to say, once you've rebasd you'll need to run:

```yarn mdfix-all ; yarn mdxfix-all```

That will update any reference to the migrated docs. It will become a part of the mgiration script when #207 lands but for now it's a manual process.

kabalin commented:
Thanks @andrewnicols , I addressed all the comments and fixed the cross-links. Not sure if I got your suggestion about `PAGE` correctly, please comment if more changes are needed.

andrewnicols reviewed (approved):
Looks good to me.

There's always going to be improvements that we can make here, but I think that they can happen _after_ we get documents migrated now.
